### PR TITLE
update scalafmt, same as in akka/akka

### PR DESCRIPTION
**Pull request by patriknw**
* enforce formatting by checking in travis

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.3.1")
 
 addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.2.0")
-addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.0-RC5")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.0.6")
```
Comment (raboof) [inline project/plugins.sbt]:
```suggestion
addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.3.0")
```
on this version I didn't see 'search state exploded'

File: .scalafmt.conf [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
-rewrite.rules              = [RedundantParens, SortImports]
-spaces.inImportCurlyBraces = true
+rewrite.rules              = [RedundantParens, SortImports, AvoidInfix]
 unindentTopLevelOperators  = true
-//binPack.callSite           = true
+align.tokens               = [{code = "=>", owner = "Case"}]
+align.openParenDefnSite    = false
+align.openParenCallSite    = false
+optIn.breakChainOnFirstMethodDot = false
+optIn.configStyleArguments = false
```
Comment (raboof) [inline .scalafmt.conf]: TBQH I'm not sure if I like this one - this is what forces
```
      val checks = im.Seq(
        SlowCheck,
        OkCheck
      )
```
to be rewritten as
```
      val checks = im.Seq(SlowCheck, OkCheck)
```
While setting this to 'true' seems to allow either (does not touch it)

File: .travis.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ jobs:
     script:
     - jabba use "adopt@1.8-0"
     - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
-    - sbt -jvm-opts .jvmopts-travis ";+test:compile;docs/paradox"
+    - sbt -jvm-opts .jvmopts-travis ";scalafmtCheckAll;+test:compile;docs/paradox"
     - git diff --exit-code --color || { echo "[error] Unformatted code found. Please run 'test:compile' and commit the reformatted code."; false; }
```
Comment (raboof) [inline .travis.yml]: This line is now no longer necessary
```suggestion
```

Comment (patriknw) [reply .travis.yml]: removed (would be needed if we add javafmt)

File: .scalafmt.conf (hunk 2 above)
Comment (patriknw) [reply .scalafmt.conf]: I agree, it seems to depend on where the `)` is placed

File: project/plugins.sbt (hunk 1 above)
Comment (patriknw) [reply project/plugins.sbt]: good, updated

File: .scalafmt.conf (hunk 2 above)
Comment (raboof) [reply .scalafmt.conf]: you could
```
git reset HEAD^
git add .scalafmt.conf project/plugins.sbt
git restore .
sbt scalafmtAll
git commit -a
```
for a smaller diff now ;)

Comment (patriknw) [reply .scalafmt.conf]: I think I did something like that to reformat all with the new version/settings. Do you see anything unexpected?

Comment (raboof) [reply .scalafmt.conf]: I would not expect https://github.com/akka/akka-management/pull/632/files#diff-b7dade6df5a2268bddde162140416a7fR161 anymore

Comment (patriknw) [reply .scalafmt.conf]: ah, I tried to change those manually. If placing the trailing `)` on a new line (as it was) then it keeps them as it. I was thinking that it looked better with one-liners for these, and misunderstood that you had the opposite opinion. I'll revert those manual changes.

Comment (raboof) [reply .scalafmt.conf]: Ahh gotcha. I don't have a strong opinion in these particular cases, but in general I like scalafmt not to touch those when I do :smile:

File: .travis.yml [hunk 4]
```diff
@@ -36,8 +36,7 @@ jobs:
     script:
     - jabba use "adopt@1.8-0"
     - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
-    - sbt -jvm-opts .jvmopts-travis ";+test:compile;docs/paradox"
-    - git diff --exit-code --color || { echo "[error] Unformatted code found. Please run 'test:compile' and commit the reformatted code."; false; }
+    - sbt -jvm-opts .jvmopts-travis ";scalafmtCheckAll;+test:compile;docs/paradox"
```
Comment (ignasi35) [inline .travis.yml]: Consider `+scalafmtCheckAll scalafmtSbtCheck` in case you have code dependant on the scala version and/or sbt code.

**Pull request merged**